src/tracer.h

```diff
@@ -62,7 +62,16 @@ class Tracer : public ot::Tracer, public std::enable_shared_from_this<Tracer> {
   void Close() noexcept override;
 
  private:
-  ot::expected<void> inject(const ot::SpanContext &sc, const ot::TextMapWriter &writer) const;
+  template <class Writer>
+  ot::expected<void> inject(const ot::SpanContext &sc, Writer &writer) const try {
```

Comment:
Could we put this definition in the cpp?

Reply:
Yeah, but it can make a wider mess.
(There might be a clean way to do it, but I've only seen messy ways)